osu.Game/Beatmaps/BeatmapSetInfo.cs

```diff
@@ -61,7 +62,8 @@ public class BeatmapSetInfo : IHasPrimaryKey, IHasFiles<BeatmapSetFileInfo>, ISo
 
         public string Hash { get; set; }
 
-        public string StoryboardFile => Files.Find(f => f.Filename.EndsWith(".osb", StringComparison.OrdinalIgnoreCase))?.Filename;
+        [JsonIgnore]
+        public string StoryboardFile => ((IBeatmapSetInfo)this).StoryboardFile;
```

Comment:
This can't ever work, the getter is just calling itself. It's a current [weird corner case of default implementations](https://stackoverflow.com/questions/59387322/making-member-virtual-prevents-calling-default-interface-implementation-and-caus) (thanks for @jai-x for actually finding this SO post, but I got to the same conclusions via [sharplab](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAEMEDZtyYDCmA3qppdlmAHbACmATgGYCGAxg5gJIAqDAM7AyFKuIzYEABkwARBuwCuEYDwC2ABwgN1Dem2BgA9rUwBeAHyYARABNFbFcBsBuMZQC+HzD8nxMAWFMEF4gkXIUcQksHFkFZVUNbV19YEMTMytMAAoc/iFgAEpgAAswQSKAOgSnJK0dPQMjU3coqm92yj84QkjoqhwAThzaBgB3QMKc6trnZMa0jNMitvFOzyA==) - note that the `virtual get_DefaultImplementation` on `Test` is calling itself via `callvirt`).

Viable routes are: renaming the default implementation in `IBeatmapSetInfo` or moving to extension methods. I have a bit of a dislike towards default interface implementations so I'd prefer an extension but I'll wait for @smoogipoo to weigh in here.

Reply:
I'm probably going to move these to extension methods. I really dislike the interface default usage requiring cast (or sometimes `this.`?) to make work.

Reply:
There was only one usage of this method so I've inlined it instead.